furbaby/api/views.py

```diff
@@ -177,3 +187,125 @@ def password_reset_token_created(sender, instance, reset_password_token, *args,
     )
     msg.attach_alternative(email_html_message, "text/html")
     msg.send()
+
+
+@api_view(["GET", "POST", "OPTIONS", "PUT", "PATCH", "DELETE"])
+def user_location_view(request):
+    location_view = UserLocationView()
+
+    if not request.user.is_authenticated:
+        return json_response({"isAuthenticated": False}, status=status.HTTP_401_UNAUTHORIZED)
+
+    # fetch all user locations for the user
+    if request.method == "GET":
+        locations_list = location_view.get_user_locations(request.user.id)
+        return json_response(locations_list, status=status.HTTP_200_OK, safe=False)
+
+    # insert a new location record for the user
+    if request.method in ["POST"]:
+        request.data["user_id"] = request.user.id
+        return location_view.insert_location_record(request.data)
```

Comment:
```suggestion
        return location_view.insert_location_record(request.data, request.user.id)
```

Don't `mutate` the `data` object, please? Since it's the request data we're dealing with, we shouldn't really be performing updates on it or adding new fields to it.

You could also pass the `request` as an argument to your function as well in case you don't want to increase the number of args.